my_enumerable.rb

```diff
@@ -0,0 +1,185 @@
+# rubocop: disable Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    each do |x|
+      yield(x)
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    idx = 0
+    while idx < to_a.length
+      yield(to_a[idx], idx)
+      idx += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    select_arr = []
+    my_each { |value| select_arr << value if yield(value) }
+    select_arr
+  end
+
+  def my_all?(arg = nil)
+    if block_given?
+      to_a.my_each { |value| return false if yield(value) == false }
+      return true
+    elsif arg.nil?
+      to_a.my_each { |value| return false if value.nil? || value == false }
+    elsif !arg.nil? && (arg.is_a? Class)
+      to_a.my_each { |value| return false unless [value.class, value.class.superclass].include?(arg) }
+    elsif !arg.nil? && arg.class == Regexp
+      to_a.my_each { |value| return false unless arg.match(value) }
+    else
+      to_a.my_each { |value| return false if value != arg }
+    end
+    true
+  end
+
+  def my_any?(arg = nil)
+    if block_given?
+      to_a.my_each { |idx| return true if yield(idx) }
+      return false
+    elsif arg.nil?
+      to_a.my_each { |idx| return true if idx }
+    elsif !arg.nil? && (arg.is_a? Class)
+      to_a.my_each { |idx| return true if [idx.class, idx.class.superclass].include?(arg) }
+    elsif !arg.nil? && arg.class == Regexp
+      to_a.my_each { |idx| return true if arg.match(idx) }
+    else
+      to_a.my_each { |idx| return true if idx == arg }
+    end
+    false
+  end
+
+  def my_none?(arg = nil, &block)
+    !my_any?(arg, &block)
+  end
+
+  def my_count(arg = nil)
+    count = 0
+    if arg.nil? && block_given?
+      my_each { |value| count += 1 if yield(value) }
+    elsif arg.nil?
+      my_each { |_value| count += 1 }
+    else
+      my_each { |value| count += 1 if value == arg }
+    end
+  end
```

Comment:
* [ ] should have working `my_count` identical and returns the same thing as ruby's [`count`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-count). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_count.png) from [Odin](https://github.com/TheOdinProject/curriculum/blob/master/ruby_programming/archive/basic_ruby/project_advanced_building_blocks.md#project-2-enumerable-methods))

#### my_count
> should work with ranges and hashes too

- [ ] my_count returns the number of items in enum through enumeration
```ruby
[5, 4, 1].my_count == [5, 4, 1].count => 3
```
- Yours is giving a LocalJumpError
![image](https://user-images.githubusercontent.com/52209067/95802252-479f3a80-0cba-11eb-8e55-eeb12fe0af67.png)

- Real count
![image](https://user-images.githubusercontent.com/52209067/77696292-c200ab00-6f72-11ea-8e97-5a2b79256cbc.png)

- [ ] my_count counts the number of items in enum that are equal to the item if an argument is given
```ruby
[5, 4, 1].my_count(4) == [5, 4, 1].count(4) => 1
```
- Yours is giving an ArgumentError
![image](https://user-images.githubusercontent.com/52209067/95802325-656c9f80-0cba-11eb-8c92-05b77024704d.png)

- Real count
![image](https://user-images.githubusercontent.com/52209067/77696378-e2306a00-6f72-11ea-9310-0bb30ab9c94d.png)